client/src/screens/createTraining/controller.js

```diff
@@ -3,9 +3,30 @@ export const controller = (serviceLocator) => {
   const { trainingsApi, navigationApi } = serviceLocator.getAPIs();
 
   const getData = () => getStoreData(controller.storeDataAccessors);
+  const createSetsPreview = (sets) => {
+    let setsPreview = '';
+
+    sets.forEach((set, index) => {
+      setsPreview += `${set.repetitions}x${set.weight}kg`;
+      if (index < (sets.length - 1)) setsPreview += ' - ';
+    });
+
+    return setsPreview;
+  };
 
   return {
-    getData: () => getData().newTraining,
+    getData: () => {
+      const newTraining = getData().newTraining;
+
+      if (newTraining !== null) {
+        newTraining.exercises = newTraining.exercises.map(exercise => ({
+          ...exercise,
+          setsPreview: createSetsPreview(exercise.sets),
+        }));
```

Comment:
the name for the bad thing that's going on here is - **mutation** (you mutate `newTraining`).
instead, it's better to do the next way:
```
if (newTraining !== null) {
  return {
     ...newTraining,
     exercises: newTraining.exercises.map(exercise => ({
          ...exercise,
          setsPreview: createSetsPreview(exercise.sets),
        })),
  }
}
```